OpenWeather/Program.cs

```diff
@@ -123,12 +126,22 @@ private static IOmfIngressService ConfigureAdhOmf(Uri address, string tenantId,
         /// </summary>
         private static string HttpGet(string url)
         {
-            Uri uri = new (url);
-            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
-            using HttpWebResponse response = (HttpWebResponse)request.GetResponse();
-            using Stream stream = response.GetResponseStream();
-            using StreamReader reader = new (stream);
-            return reader.ReadToEnd();
+            using HttpRequestMessage request = new (HttpMethod.Get, url);
+            return Send(request).Result;
+        }
+
+        /// <summary>
+        /// Send message using HttpRequestMessage
+        /// </summary>
+        private static async Task<string> Send(HttpRequestMessage request)
```

Comment:
if you haven't already, I would consider passing a HttpMethod to Send to avoid having a method per method. Just a suggestion so feel free to disregard